tests/unit/test_rose_stem_units.py

```diff
@@ -54,3 +55,48 @@ def test_get_source_opt_from_args(tmp_path, monkeypatch, args, expect):
         assert result == expect
     else:
         assert result == expect.format(tmp_path=str(tmp_path))
+
+
+class Test_generate_name:
+    """Try 3 paths in StemRunner._generate_name.
+
+    Contains a setup function for each path.
+    """
+    EXPECT = 'bar'
+    MSG = 'Suite is named bar'
+
+    @staticmethod
+    def name_from_fcm(monkeypatch):
+        """It gets the name from _ascertain_project."""
+        class MonkeyStemRunner(StemRunner):
+            def _ascertain_project(self, x):
+                return ['foo', '/foo/bar']
+        return MonkeyStemRunner(SimpleNamespace, reporter=print)
+
+    @staticmethod
+    def source_set(monkeypatch):
+        """It gets the name from the source opt."""
+        opts = SimpleNamespace()
+        opts.source = 'bar/rose-stem'
```

Comment:
`/foo/bar/rose-stem` might be a slightly better test?

Reply:
```suggestion
        opts.source = '/foo/bar/rose-stem'
```